### PR TITLE
move the getEntityManager, only get it if needed

### PR DESCRIPTION
**Pull request by OskarStark**
if this will be merged, please be sure, that this is the correct branch!

Thank you

### CODE REVIEW

File: cookbook/doctrine/event_listeners_subscribers.rst [hunk 1]
```diff
@@ -135,10 +135,10 @@ a ``postPersist`` method, which will be called when the event is dispatched::
         public function postPersist(LifecycleEventArgs $args)
         {
             $entity = $args->getEntity();
-            $entityManager = $args->getEntityManager();
 
             // perhaps you only want to act on some "Product" entity
             if ($entity instanceof Product) {
```
Comment (timglabisch) [inline cookbook/doctrine/event_listeners_subscribers.rst]: i would prefer returning early.

if (!($entity instanceof Product)) { return; }

Comment (OskarStark) [inline cookbook/doctrine/event_listeners_subscribers.rst]: this is not so clear IMO, maybe you want to check for another Type later?

Comment (WouterJ) [inline cookbook/doctrine/event_listeners_subscribers.rst]: While the guard style is nice, I think the current syntax is more clear for people not familair with this style of guarding (and I assume lots of beginners are not familair with this syntax).

**Comment by WouterJ** [comment]
:+1: 

I think we should add a new line before the `// ...` comment, but that's such a minor thing that it can be done while merging.

**Comment by weaverryan** [comment]
Looks fine to me :+1:

File: cookbook/doctrine/event_listeners_subscribers.rst (hunk 1 above)
Comment (weaverryan) [inline cookbook/doctrine/event_listeners_subscribers.rst]: I generally would be pro "returning early" - i.e. reversing the logic here. I see too many nested if statements in people's code - teaching them to return is a tiny win.

Comment (xabbuh) [inline cookbook/doctrine/event_listeners_subscribers.rst]: I am in favour of return early here too.

**Comment by WouterJ** [comment]
Thanks Oskar.

File: cookbook/doctrine/event_listeners_subscribers.rst (hunk 1 above)
Comment (WouterJ) [inline cookbook/doctrine/event_listeners_subscribers.rst]: See fab0985

**Pull request closed without merging**